src/components/app.tsx

```diff
@@ -17,35 +17,68 @@ import VizPane from './viz-pane';
 type Props = ReturnType<typeof mapDispatchToProps>;
 
 class App extends React.Component<Props & {match: any, location: any}> {
+  static data: any;
+  static evt: any;
   public componentDidMount() {
-    window.addEventListener(
-      'message',
-      (evt) => {
-        const data = evt.data;
-        if (!data.spec) {
-          return;
-        }
-        console.info('[Vega-Editor] Received Message', evt.origin, data);
-        // send acknowledgement
-        const parsed = JSON.parse(data.spec);
-        data.spec = JSON.stringify(parsed, null, 2);
-        if (data.spec || data.file) {
-          evt.source.postMessage(true, '*');
-        }
-        if (data.spec) {
-          if (data.mode.toUpperCase() === 'VEGA') {
-            this.props.updateVegaSpec(data.spec);
-          } else if (data.mode.toUpperCase() === 'VEGA-LITE') {
-            this.props.updateVegaLiteSpec(data.spec);
-          }
-        }
-      },
-      false,
-    );
-
+    window.addEventListener('message', (evt) => this.eventListener(evt), false);
     const parameter = this.props.match.params;
     this.setSpecInUrl(parameter);
   }
+  /**
+   * Listener of event `message`
+   */
+  public eventListener(evt): any {
+    if (!evt.data) {
+      return;
+    }
+    App.evt = evt;
+    App.data = evt.data;
+    console.info('[Vega-Editor] Received Message', evt.origin, App.data);
+    // send acknowledgment
+    const parsed = JSON.parse(App.data.spec);
+    const dataURL = parsed['data']['url'];
+
+    // Checking wheather link is absolute or relative to editor
+    this.loaderPromise({baseURL: 'https://vega.github.io/editor/'}, dataURL).then((data) => {
```

Comment:
I think we should first check whether the URL is absolute. In that case, we won't do anything.

Reply:
`vega-loader` already checks whether it absolute or not. If not then I am checking it with relative to the editor which means we aren't doing anything when URL is absolute. I tested it.

Reply:
Okay, this will might change when we switch to a custom loader.